from werkzeug.exceptions import NotFound
from ..db.models import User, EmailConfirmationToken
from .auth import auth


def signup(body):
    # TODO validate json input
    # TODO check if user details already exist
    user = User.create(body.get('username'), body.get('email'),
                       body.get('password'))
    return user.to_obj(whos_asking=user)


def confirm_email(token):
    try:
        EmailConfirmationToken.confirm(token)
    except Exception as e:
        raise NotFound(str(e))

    return {'confirmed': True}


@auth
def detail(id, auth_user):
    detail_user = User.query.filter_by(id=id).first()
    return detail_user.to_obj(whos_asking=auth_user)
